### PR TITLE
Additional gateway details

### PR DESCRIPTION
**Pull request by barnjamin**

### CODE REVIEW

File: docs/reference/components/gateway.md [hunk 1]
```diff
@@ -5,49 +5,86 @@ Note: This page is a work in progress, check back for updates
...
-## Integration
+## Details
 
-Integration with Wormhole Gateway can be accomplished with a few lines of code.
+The Wormhole Gateway is implemented as a set of contracts and modules.
 
-<!-- TODO: change branch to `main` once merged -->
+### Wormhole Core Contracts
+
+The [core contracts](./core-contracts.md) to emit messages and verify [Guardian](./guardian.md) signatures are still required on each Cosmos chain.
```
Comment (nik-suri) [inline docs/reference/components/gateway.md:23]: This is only required if that Cosmos chain wants to integrate generic messaging.

For token bridging, Cosmos chains don't need to do anything.

File: docs/reference/components/gateway.md [hunk 2]
```diff
@@ -5,49 +5,86 @@ Note: This page is a work in progress, check back for updates
...
+- Mints [Token Factory](#token-factory-module) tokens
+- Decodes the additional payload as a [`GatewayIbcTokenBridgePayload`](#gatewayibctokenbridgepayload)
+- Sends tokens via IBC to destination cosmos chains
+
+The contract also supports transfers _out of_ the Cosmos ecosystem by implementing an `execute` handler which accepts a `GatewayIbcTokenBridgePayload` and an amount of tokenfactory tokens in `info.funds` (which are the tokens to be bridged out). 
+
+- Burn the [Token Factory](#token-factory-module) tokens 
+- Send the corresponding CW20 tokens to the [Token Bridge](./core-contracts.md#token-bridge) via IBC  
+<!-- TODO: what calls this? TokenBridge or source contract? -->
+- Calls `InitiateTransfer` or `InitiateTransferWithPayload` based on whether the [`GatewayIbcTokenBridgePayload`](#gatewayibctokenbridgepayload) is of type `Simple` or `ContractControlled`
```
Comment (nik-suri) [inline docs/reference/components/gateway.md:41]: Correct flow is:
- Burn the token factory tokens (as you mention)
- Unlock the cw20 tokens
- Grants the token bridge approval to spend the cw20 tokens
- Calls initiateTransfer or initiateTransferWithPayload (as you mention)

If you have wh tokens on a Cosmos chain and you want to bridge them out, you send them via IBC with the memo formatted as a `gateway_ibc_token_bridge_payload `. The IBC Composability middleware handles properly routing to either an external chain or another cosmos chain

File: docs/reference/components/gateway.md [hunk 3]
```diff
@@ -5,49 +5,86 @@ Note: This page is a work in progress, check back for updates
...
 
-To bridge assets into a Cosmos chain, an asset transfer is initiated on the foreign chain with a [payload](https://github.com/wormhole-foundation/wormhole/blob/gateway-integration/cosmwasm/contracts/ibc-translator/src/msg.rs#L73-L86) that is understood by the Gateway. Once received by the Gateway, the asset's CW20 representation is sent to the destination chain through IBC using the well established [ICS20 protocol](https://github.com/cosmos/ibc/tree/main/spec/app/ics-020-fungible-token-transfer).
 
-To bridge assets out of the Cosmos ecosystem or between Cosmos chains, an IBC transfer is initiated on the source chain to the Gateway with a payload containing details about the transfer.
+### Token Factory Module
 
-In short:
+A deployment of the canonical [Token Factory](https://github.com/osmosis-labs/osmosis/tree/main/x/tokenfactory) module on Wormchain.
```
Comment (nik-suri) [inline docs/reference/components/gateway.md]: We are actually using this version - https://github.com/CosmosContracts/juno/tree/v14.1.1/x/tokenfactory
